src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1036,8 +1037,9 @@ private String updateAliasQuery(){
         baseQuery.append(" ").append("WHERE");
 
         if(allowedLocations.isEmpty()){
+            String jcrSystemPath = StringUtils.removeEnd(JCR_SYSTEM_PREFIX, "/");
```

Comment:
The check performed by http://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#removeEnd-java.lang.String-java.lang.String- is not really necessary here, because you know that the constant ends with "/", right? I would just use `jcrSystemPath.substring(0, jcrSystemPath.length() - 1);` in this case.

Reply:
i would like to keep it like this as the code is just better readable this way. this code line is executed once on startup of the bundle, not need to over-optimize it here.